sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs

```diff
@@ -637,34 +638,50 @@ internal static StorageResourceItemProperties ToResourceProperties(this BlobItem
                 properties: properties);
         }
 
+        private static string ConvertContentPropertyObjectToString(object contentPropertyValue)
+        {
+            if (contentPropertyValue is string)
+            {
+                return Convert.ToString(contentPropertyValue);
+            }
+            else if (contentPropertyValue is string[])
+            {
+                return string.Join(",", (string[])contentPropertyValue);
+            }
+            else
+            {
+                throw Errors.UnexpectedPropertyType(DataMovementConstants.ResourceProperties.ContentType, DataMovementConstants.StringTypeStr, DataMovementConstants.StringArrayTypeStr);
+            }
+        }
+
         private static BlobHttpHeaders GetHttpHeaders(
             BlobStorageResourceOptions options,
             Dictionary<string, object> properties)
             => new()
             {
                 ContentType = (options?.ContentType?.Preserve ?? true)
                     ? properties?.TryGetValue(DataMovementConstants.ResourceProperties.ContentType, out object contentType) == true
-                        ? (string) contentType
+                        ? (contentType is string) ? Convert.ToString(contentType) : throw Errors.UnexpectedPropertyType(DataMovementConstants.ResourceProperties.ContentType, DataMovementConstants.StringTypeStr)
```

Comment:
do we just want to call `ConvertContentPropertyObjectToString` for all these? It should do the same thing?

Reply:
reverted back.